_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp

```diff
@@ -237,16 +229,19 @@ mfxStatus MFXVideoENCODEVP9_HW::Init(mfxVideoParam *par)
     }
     else if (m_video.IOPattern == MFX_IOPATTERN_IN_OPAQUE_MEMORY)
     {
-        mfxExtOpaqueSurfaceAlloc &opaq = GetExtBufferRef(m_video);
+        mfxExtOpaqueSurfaceAlloc& opaq = GetExtBufferRef(m_video);
+        request.Type = opaq.In.Type;
+        request.NumFrameMin = opaq.In.NumSurface;
 
-        sts = m_pCore->MapOpaqueSurface(m_pCore->pthis, opaq.In.NumSurface, opaq.In.Type, opaq.In.Surfaces);
+        sts = m_opaqFrames.Init(m_pCore, &request);
         MFX_CHECK_STS(sts);
 
         if (opaq.In.Type & MFX_MEMTYPE_SYSTEM_MEMORY)
         {
-            request.NumFrameMin = request.NumFrameSuggested = opaq.In.NumSurface;
+            request.Type = MFX_MEMTYPE_D3D_INT;
+            request.NumFrameMin = opaq.In.NumSurface;
```

Comment:
shouldn't we set NumFrameSuggested ?

Reply:
'NumFrameSuggested' not used for allocation if fact. I'm removing NumFrameSuggested for all cases here to make it more clear.